src/app-locator.js

```diff
@@ -0,0 +1,98 @@
+import appIds from './known-app-ids'
+
+// Known apps and their assigned ports
+const DEFAULT_LOCAL_URLS = {
+  [appIds['Voting']]: 'http://localhost:3001/',
+  [appIds['Finance']]: 'http://localhost:3002/',
+  [appIds['TokenManager']]: 'http://localhost:3003/',
+  [appIds['Survey']]: 'http://localhost:3004/',
+  [appIds['Agent']]: 'http://localhost:3005/',
+  [appIds['Fundraising']]: 'http://localhost:3006/',
+}
+
+// Split a single bridge declaration into an app + a location
+function splitBridge(bridge) {
+  const firstColon = bridge.indexOf(':')
+  if (firstColon === -1) {
+    return [bridge]
+  }
+  return [
+    bridge.slice(0, firstColon).trim(),
+    bridge.slice(firstColon + 1).trim(),
+  ]
+}
+
+function getAppId(app) {
+  // known app name
+  if (appIds[app]) {
+    return appIds[app]
+  }
+
+  // probably a valid app ID
+  if (app.startsWith('0x') && app.length === 66) {
+    return app
+  }
+
+  return null
+}
+
+function getAppUrl(appId, location) {
+  // get the default URL for this appId (if location is local)
+  if ((!location || location === 'local') && DEFAULT_LOCAL_URLS[appId]) {
+    return DEFAULT_LOCAL_URLS[appId]
+  }
+
+  // probably a valid HTTP URL
+  if (location.startsWith('http')) {
```

Comment:
We could also check for just `startsWith('localhost')` too; it's pretty common to just use `localhost:3000` or etc.

Reply:
Maybe we could do it the other way? It could be useful for local network IPs, or for domains pointing to 127.0.0.1.

If something is not:

- Empty.
- Special (`ipfs`, `local`).
- A port.
- A full URL (starting with `http(s)://` for now, and maybe we could also support `ipfs:xyz` later).

Then it is a domain or an IP address, and we prefix it with `http://`.

Reply:
That sounds good!

Reply:
Done in https://github.com/aragon/aragon/pull/1124/commits/71cc90c052c9b735ad719e3c0fe14b0c01843478